apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -9,58 +9,54 @@ import {
   getFooter,
   getFullPageById,
   getNavBar,
+  updateBlobById,
+  updatePageById,
 } from '../resource/resource.service'
 import { getSiteConfig } from '../site/site.service'
 
+// NOTE: Need to do validation like checking for existence of the page
+// and whether the user has write-access to said page
+const pageProcedure = protectedProcedure
+
 export const pageRouter = router({
-  readPageAndBlob: protectedProcedure
+  readPageAndBlob: pageProcedure
     .input(getEditPageSchema)
     .query(async ({ input, ctx }) => {
       const { pageId, siteId } = input
       const page = await getFullPageById(pageId)
-      // TODO: Fill these in later
       const pageName: string = page.name
+      // NOTE: Sitemap in siteMeta is immediate parent, immediate children and siblings
       const siteMeta = getSiteConfig(siteId)
       const navbar = getNavBar(siteId)
       const footer = getFooter(siteId)
       const { content } = page
 
       return {
         pageName,
-        // NOTE: might shift theme, isGovt, navbar, footer out into separate function?
-        // because this is shared across the whole site (site level props)
-        ...siteMeta,
         navbar,
         footer,
-        // NOTE: This is immediate parent, immediate children and siblings
         content,
+        ...siteMeta,
       }
     }),
 
-  updatePage: protectedProcedure
+  updatePage: pageProcedure
     .input(updatePageSchema)
     .query(async ({ input, ctx }) => {
-      const parentId = ''
-      const pageName = ''
+      await updatePageById({ ...input, id: input.pageId })
 
-      return {
-        // NOTE: This should adhere to our db schema
-        parentId,
-        pageName,
-        // NOTE: Don't need `variant` here
-        // because the router itself discriminates for us
-      }
+      return input
     }),
-  updatePageBlob: protectedProcedure
+
+  updatePageBlob: pageProcedure
     .input(updatePageBlobSchema)
     .query(async ({ input, ctx }) => {
-      // NOTE: Not returning the `content` first because
-      // 1. it might potentially be huge
-      // 2. frontend already knows
-      // ahead of time what is the content (it sent the content over)
-      return {}
+      await updateBlobById({ ...input, id: input.pageId })
+
+      return input
```

Comment:
will it result in unnecessary network activities, or is there a use case that requires the input back

Reply:
actually i prefer not returning but change was made due to comment [here](https://github.com/opengovsg/isomer/pull/222#discussion_r1651182789). 

i don't have any real concerns now, i think we will find out when pages start getting big